### DISCUSSION TITLE
troubleshooting libltdl.so.7

### DISCUSSION THREAD

**Discussion by nicopiot** [fat-tire/resolve#56] [Q&A]
Hello, thanks for all the great work. 
I am new to podman, I followed all instructions and everything seemed to work, installing on a linux mint 21.3. 
when I try ./resolve.sh, I get the following : 

```
non-network local connections being added to access control list
------------------------------------------------------------------------------
The container's /etc/machine-id  : e0fb6a2bfeb84a18848d9a9795f21d7c
The network driver setting is          : --network=host
Bind-mounted directories of interest   :
  CONTAINER (CentOS 8)                  -> HOST (Linux Mint)
  ----------------------------------------------------------------------------
  /home/resolve                         -> /home/famille/containers/resolve/mounts/resolve-home
  /opt/resolve/logs                     -> /home/famille/containers/resolve/mounts/logs
  /opt/resolve/configs                  -> /home/famille/containers/resolve/mounts/configs
  /opt/resolve/easyDCP                  -> /home/famille/containers/resolve/mounts/easyDCP
  /opt/resolve/.license                 -> /home/famille/containers/resolve/mounts/license
  /opt/resolve/'Resolve Disk Database'  -> /home/famille/containers/resolve/mounts/database
  /var/BlackmagicDesign/DaVinci Resolve -> /home/famille/containers/resolve/mounts/BlackmagicDesign
  /opt/resolve/Media                    -> /home/famille/containers/resolve/mounts/Media
/opt/resolve/bin/resolve: error while loading shared libraries: libltdl.so.7: cannot open shared object file: No such file or directory
```
I tried in the container to do `sudo yum install libltdl.so.7`
again, it installed three packages, including libtool if I remember correctly the name, but doesn't help. 

Can anyone help me with this? 

I also tried to install directly resolve on linux mint, it worked but it doesn't lauch more than the spashscreen, so I believe the container is the way to go. 

Many thanks

Nicolas

**Comment by nicopiot** [comment]
I found what happened on the yum install so I paste it here if it helps : 
```
bash-4.4$ sudo yum install libltdl.so.7
Rocky Linux 8 - AppStream                                                                                   1.0 MB/s |  11 MB     00:11    
Rocky Linux 8 - BaseOS                                                                                      951 kB/s | 6.0 MB     00:06    
Rocky Linux 8 - Extras                                                                                       31 kB/s |  14 kB     00:00    
Last metadata expiration check: 0:00:01 ago on Sat Aug 10 19:29:26 2024.
Dependencies resolved.
============================================================================================================================================
 Package                                Architecture              Version                                 Repository                   Size
============================================================================================================================================
Installing:
 libtool-ltdl                           i686                      2.4.6-25.el8                            baseos                       59 k
Installing dependencies:
 glibc                                  i686                      2.28-251.el8_10.2                       baseos                      2.0 M
Installing weak dependencies:
 glibc-gconv-extra                      i686                      2.28-251.el8_10.2                       baseos                      1.6 M

Transaction Summary
============================================================================================================================================
Install  3 Packages

Total download size: 3.6 M
Installed size: 11 M
Is this ok [y/N]: y

```

**Comment by nicopiot** [comment]
also trid to build with the 18.6.6 da vinci resolve, even worst, it tells me `The 'resolve' image wasn't found.  Build the Dockerfile first.
`

**Comment by fat-tire** [comment]
> /opt/resolve/bin/resolve: error while loading shared libraries: libltdl.so.7: cannot open shared object file: No such file or directory

So I am running in ubuntu 24.04 LTS.  I am able to run successfully without any dependency warning.  Moreover I don't see an explicit requirement for it:

```bash
$ ldd  /opt/resolve/bin/resolve | grep libtld.so.7
$ 
```

(shows nothing) 

In fact I searched the whole container for ` libtld.so.7` and can't find it anywhere.

The only libraries that contain ".7" that are required are:

```
	libswscale.so.7 => /opt/resolve/bin/../libs/libswscale.so.7 (0x000071c495600000)
	libffi.so.7 => /opt/resolve/bin/../libs/libffi.so.7 (0x000071c48a400000)
```

If you try the `ldd` command on the binary, do you see the missing library?  I did find [some](https://github.com/grtjn/mlvagrant/issues/132) [related](https://stackoverflow.com/questions/45121945/jenkins-in-docker-container-run-docker-pipeline)  [threads](https://stackoverflow.com/questions/47782554/what-is-the-difference-between-libdl-and-libltdl) that might be worth looking at.

Which version specifically of the .zip did you download, and what is your hardware?  Are you running Linux Mint directly on the hardware or via some kind of VM or emulator?

It looks like this file is in libtool-ltdl, so, what was the result when you tried installing?  I'm assuming it didn't work, but did the error change at all?

```bash
$ dnf install libtool-ltdl.x86_64
```

If that doesn't put the file in `/usr/lib64/libtldl.so.7`, you can find an appropriate package [here](https://rpmfind.net/linux/rpm2html/search.php?query=libltdl.so.7()(64bit)) using this random site I just found :)

Remember that with a container, once you exit, anything you installed won't be there the next time, unless you change the `Dockerfile` to include it and then rebuild.


FWIW I see here that it [WAS required back in 2018](https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=56878&start=750#p388072).  Are you using a recent .zip?

Hope something here will be helpful.

**Comment by nicopiot** [comment]
Hello, and thanks for the quick answer, much appreciated.


first I installed with DaVinci_Resolve_Studio_17.0_Linux.zip

then, I realised (when I tried to install straight on mint) that this is the pay (studio) version, so i searched for a free one, but only found the latest from blackmagic which is DaVinci_Resolve_18.6.6_Linux.zip

the 18.6.6 install straight on linux mint doesn't work, I get stuck on the first splash screen, so the centos container is definitely the way to go

So I tried to build with this 18.6.6 zip, but it's even worst, install process completes fine, no problem, but then it doesn't launch at all, saying I need to do the dockerfile first.


BTW, I run my linux mint directly on the hardware. It seems the GPU nvidia is well recognised, as per the test you suggest to do.

I just downloaded a 17.4.6 version, and tried with that. no good : 
```
Copied libs/libQt5Concurrent.so.5
Error: Installer failed
 Failed to copy file: libs/libcudnn_cnn_infer.so.8
Blackmagic Design Pty. Ltd.
License Agreement for DaVinci Resolve
```
and at the end : 
```
STEP 17/18: RUN if [[ "${BUILD_X264_ENCODER_PLUGIN}" == 1 ]] ; then        if [[ `dnf repolist --all` == *"crb"* ]]; then export POWERTOOLS=crb ; fi        && cd /tmp        && sudo dnf -y install clang llvm zlib-devel git diffutils patch ed        && sudo dnf -y --enablerepo="${POWERTOOLS}" install nasm        && git clone https://code.videolan.org/videolan/x264.git        && cd x264        && ./configure --enable-shared        && make        && cp -R /opt/resolve/Developer/CodecPlugin/Examples/x264_encoder_plugin /tmp        && cd ..        && chmod a+x x264_plugin_patcher.sh        && ./x264_plugin_patcher.sh        && cd x264_encoder_plugin && make clean && make && make install        && dnf remove -y clang llvm zlib-devel git diffutils patch nasm ed        && rm -rf /var/cache/yum/*        && dnf clean all ;     fi        && rm -rf /tmp/x264*
STEP 18/18: CMD __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve
COMMIT resolve:17.4.6
Getting image source signatures
Copying blob 879054335f94 skipped: already exists  
Copying blob aa310f671ac9 done  
error committing container for step {Env:[ZIPNAME=DaVinci_Resolve_17.4.6_Linux.zip USER_ID=1000 NO_PIPEWIRE=0 POWERTOOLS=powertools NVIDIA_VERSION=535.183.01 BUILD_X264_ENCODER_PLUGIN=0 ARCH=x86_64 EXTRA_PACKS= USER=resolve PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin] Command:cmd Args:[__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve] Flags:[] Attrs:map[] Message:CMD __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve Original:CMD __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve}: error copying layers and metadata for container "62f915ebd38ba550efd50bda7c6022f9acfd49e5e90b0bd844ea2be513006f18": writing blob: adding layer with blob "sha256:aa310f671ac9f9d35e57cba2381a2270f6991b98a7a56d8105f6de25bc80716c": Error processing tar file(exit status 1): write /opt/resolve/Onboarding/translations/qtwebengine_locales/it.pak: no space left on device
ERRO[1373] exit status 125                              
Build of resolve:17.4.6 is complete.  To run resolve, try:

./resolve.sh
```
don't really know what to do from here

regards

Nicolas

**Reply by nicopiot** [reply]
it says error "no space left" but I cleaned my hard drive and there is 15 GB left on it. 
Where do I find the container files on mint? Does it add space each time I build, or does it overwrite? 
I will try to clean even more (timeshift is clogging the system since I started this process) and do it again.

**Reply by cowmix** [reply]
Is /var on its own volume, separate from your /home directory? If you are using Docker as your container system, that’s typically where all Docker-related files (images, volumes, etc.) are stored.

**Reply by fat-tire** [reply]
I only know podman but you can use:

`podman images`  (or I assume `docker images`) to see all the container images, including various ones that you may not need. You could then do `podman image prune` (or `docker image prune` I assume) to get rid of "dangling" images that aren't being actively used.  There is also a way to remove half-broken builds, but it's a podman thing I think involving `buildah` I'd have to look at the actual command..

when you type:

`df -h`

you should get a list of the various devices and how much room you have left on each one.  One thing you can do to make room also is get rid of any older `.zip` files, especially if they're sitting in your Trash just taking up space.

I think it will overwrite the images, to answer that question-- but only once the new "latest" image is built.  While it's building it can use some space-- resolve and the OS it uses isn't small...

That 18.6.6 version you found is the latest stable version, though the version 19b6 that's out has been getting pretty good reviews FWIW.

Let me know if making room gets you past the issue.. it looks to me like this is the cause of your latest problem.  Not sure what's up with the weird library that it was wanting before, but maybe you'll get past it with the 18.6.6+ versions.

Update;  A quick search says `docker builder prune -a` will get rid of the build cache, which might save you some room.  You can also use `docker rmi <image_name>` to remove a specific image that appears in your list when you do `docker images`.

**Reply by nicopiot** [reply]
Thanks for all your commitment to help me. 
At last, it's running! 
I cleared all I could on my drive (timeshift images, trashbin,) and cleared all podman stuffs with `$podman system reset`
this is the nuclear option : cleared everything to start anew.
in the dockerfile, I added  `ARG EXTRA_PACKS="libtool-ltdl.x86_64"`
Then I built the 17.4.6 version found here https://www.videohelp.com/software/DaVinci-Resolve/old-versions 
and it went well.
first launch, it was stuck on a blank (actually grey) launch screen. After a while, I killed the process, and tried again, and this time it launched. 
The shared folders are working. I imported a mp4 file that doesn't play, but it must be a codec thing. The software is running at last! I have to learn it now, coming from lightworks. 
So maybe my experience can go in the troubleshooting section of the readme file, it might help others. 
Thanks again for all. Cheers. 
Nicolas

**Comment by fat-tire** [comment]
This is terrific news!  Congrats!  You may notice some issues with sound on linux-- you just need to do a conversion to an audio format that is supported.  Not sure what the blank launch screen was about, but maybe restarting helped clear it out of whatever it was getting stuck on...

I'd love an addition to the README-- feel free to contribute a section with the probelm and your solution (I figure you can say it more accurately and completely than I can) and submit a PR!

Woo!  Glad it's working.  Enjoy and hopefully you'll find this is a big step up from lightworks.

**Reply by nicopiot** [reply]
Well, I am not yet on sound issue, as I have first a video issue : I can import all the video I want, but they just dont play : it is reading the file, I see the time moving, but no image, grey screen. I tried to convert to all format I could think of : reencode h.264 (mp4 and mov), mpg, mpeg, avi files. Nothing works. 
If I drag the file in theedit sequence, only a audia file is put on, with nothing on it! 
been searching forums on this topic, a lot of people say free version can't read h.264, only studio can while some other say h.264 is no problem for them. If you have a hint, i take it. 
So far, it's a step down from lightworks, I hope I will manage to get it to work, as everybody says it's so amazing and better! 

for the readme, can I edit it myself? I guess not. I will write up something for you and let you edit and add it to the file. I believe it is better for general coherence of everything. 

Cheers.

**Comment by fat-tire** [comment]
Gee, I'm not sure because I have the Studio version (well worth the money).  Have you tried transcoding using a different container such as .mov to see if it plays?  Also, are you able to play videos with other apps like VLC?

As for the readme-- you can fork the repository, make any additions & corrections and then when you send it back to your repository, you should see a "submit PR (Pull Request)" button.  That will allow me to review it and suggest any changes and finally when it's perfect, I can accept the pull request and it will become part of the official repository :)  Ideally you can "squish" everything into a single commit which I can accept as "changes to README to cover xyz."  In fact, you can make your changes on your own repository right from the github web site by clicking the "edit" button and making your changes (don't forget to save it)... then make sure you agree with the project license and submit that as a pull request.

The only thing I ask is that you make your additions "fit" into the current README style so that it fits in seamlessly.  But any changes you contribute will be much appreciated by everyone who uses this project, so thanks in advance!

**Reply by nicopiot** [reply]
at last, everything works, I have to reencode, but with shutter encoder, because other tries to reencode in mov container did not work. I will put all this in the md file and try to follow your explanation for the fork (I have never done that before)
